backend/src/sms/sms.service.ts

```diff
@@ -43,23 +43,64 @@ export class SmsService {
         }
     }
 
-    async sendNewSeasonSMS(recipient: Recipient, expireDate: string): Promise<boolean> {
-        const settings = SMSConfig.getTeliaConfig();
+    async sendNewSeasonSMS(recipients: Recipient[], expireDate: string): Promise<boolean> {
+        const {
+            user,
+            username,
+            password,
+            batchEndPoint,
+        } = SMSConfig.getTeliaConfig();
 
-        const message = this.getExpiredMessage(recipient.name, expireDate);
+        const batch: BatchItem[] = recipients.map(recipient => ({
+            t: recipient.phoneNumber,
+            m: this.getExpiredMessage(recipient.name, expireDate),
+        }));
 
         const messageRequest = {
-            username: settings.username, password: settings.password,
-            from: settings.user, to: [recipient.phoneNumber], message,
-        } as TeliaMessageRequest;
-        const attemptMessage = await this.sendMessageToUser(messageRequest, settings.endPoint);
+            username: username,
+            password: password,
+            from: user,
+            batch,
+        } as TeliaBatchMessageRequest;
+        const attemptMessage = await this.batchSendMessagesToUsers(messageRequest, batchEndPoint);
         if (attemptMessage) {
             return true;
         } else {
             throw new InternalServerErrorException(content.MessengerServiceNotAvailable);
         }
     }
 
+    /**
+     * Sends multiple messages in a single batch. The service withstands payloads in excess of 
+     * 100 000 individual messages per batch.
```

Comment:
I didn't find the information about the number of maximum messages in the doc. Nevertheless, if the max is 100 000 then should we handle situation where the number messages exceed this number?

Reply:
Also, please add the link to all the information you found into confluence as well. I notice there's no mentioning of the Lekab REST SMS in confluence

Reply:
This information comes from our discussion with the service producer Lekab. 100 000 is the number of messages with which the system has been **stress tested and guaranteed to work** (as per one of their customers' requirement), but they also mentioned that considerably larger batches shouldn't pose issues. In any case, this application should never be in a state where it would need to send anywhere near this amount of SMSs.

Reply:
I'll add the restSMS info to the docs. Good point!